drivers/power/ltc7841/ltc7841.c

```diff
@@ -0,0 +1,461 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "ltc7841.h"
+
+const uint8_t ltc7841_write_only_registers[WRITE_ONLY_REGISTERS_NUMBER] = {
+	LTC7841_MFR_CLEAR_PEAKS,
+	LTC7841_MFR_RESET
+};
+
+/**
+ * @brief Free the device descriptor.
+ * @param desc - The device structure.
+ * @return 0 in case of success, -EINVAL otherwise.
+ */
+int ltc7841_remove(struct ltc7841_desc *desc)
+{
+	int ret;
+	if (!desc)
+		return -EINVAL;
+
+	ret = no_os_i2c_remove(desc->comm_desc);
+	if (ret)
+		return ret;
+
+	no_os_free(desc);
+
+	return 0;
+}
+
+/**
+ * @brief Returns the size of the register, whether one byte or two bytes
+ * @param addr - Register address
+ * @return size - 1 or 2 for legitimate register address, 0 otherwise
+ */
+static uint8_t get_register_size(uint8_t addr)
+{
+	uint8_t byte_size;
+	switch (addr) {
+	case LTC7841_MFR_CLEAR_PEAKS:
+		byte_size = 0;
+		break;
+	case LTC7841_OPERATION:
+	case LTC7841_VOUT_MODE:
+	case LTC7841_PMBUS_REVISION:
+	case LTC7841_MFR_CONFIG:
+	case LTC7841_MFR_RAIL_ADDRESS:
+	case LTC7841_MFR_RESET:
+		byte_size = 1;
+		break;
+	case LTC7841_STATUS_WORD:
+	case LTC7841_READ_VIN:
+	case LTC7841_READ_VOUT:
+	case LTC7841_READ_IOUT:
+	case LTC7841_READ_TEMPERATURE_1:
+	case LTC7841_MFR_IOUT_PEAK:
+	case LTC7841_MFR_VOUT_PEAK:
+	case LTC7841_MFR_VIN_PEAK:
+	case LTC7841_MFR_TEMEPRATURE1_PEAK:
+	case LTC7841_MFR_IIN_PEAK:
+	case LTC7841_MFR_VOUT_MARGIN_HIGH:
+	case LTC7841_MFR_SPECIAL_ID:
+	case LTC7841_MFR_VOUT_COMMAND:
+	case LTC7841_MFR_VOUT_MARGIN_LOW:
+		byte_size = 2;
+		break;
+	default:
+		byte_size = 0;
+		break;
+	}
+
+	return byte_size;
+}
+
+/**
+ * @brief Initialize the device structure.
+ * @param device - The device structure to be initialized.
+ * @param init_param - Initialization parameter for the device descriptor.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ltc7841_init(struct ltc7841_desc **device,
+		 struct ltc7841_init_param *init_param)
+{
+	struct ltc7841_desc *dev;
+	int ret;
+	dev = (struct ltc7841_desc *)calloc(1, sizeof(struct ltc7841_desc));
+	if (!dev)
+		return -ENOMEM;
+	/* Initialize I2C */
+	ret = no_os_i2c_init(&dev->comm_desc, init_param->comm_param);
+	if (ret)
+		goto i2c_err;
+	/*initialize mfr vout command to half upon initializing*/
```

Comment:
try to format comments like:
```
   capital
   |
/* Text */
  ^    ^
   whitespace
```

reasoning being to have consistent looking comments, otherwise we end up with as many comment styles as contributors